### PR TITLE
8324577: [REDO] - [IMPROVE] OPEN_MAX is no longer the max limit on macOS >= 10.6 for RLIMIT_NOFILE

### PR DESCRIPTION
**Pull request by gerard-ziemski**
This is a 3rd attempt of the same fix:

1st one had to be pulled out because of a bug in zsh
2nd one had a workaround for the bug in zsh, but then uncovered an issue in JWDP (JDK-8324668), which was subsequently fixed.

Tested with MACH5 tier1-9 with no unique or new failures on macOS

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324577](https://bugs.openjdk.org/browse/JDK-8324577): [REDO] - [IMPROVE] OPEN_MAX is no longer the max limit on macOS &gt;= 10.6 for RLIMIT_NOFILE (**Enhancement** - P4)

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to [1376278e](https://git.openjdk.org/jdk/pull/18821/files/1376278e4680cd88be40df8864e5289f7330ef1f)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [1376278e](https://git.openjdk.org/jdk/pull/18821/files/1376278e4680cd88be40df8864e5289f7330ef1f)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18821/head:pull/18821` \
`$ git checkout pull/18821`

Update a local copy of the PR: \
`$ git checkout pull/18821` \
`$ git pull https://git.openjdk.org/jdk.git pull/18821/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18821`

View PR using the GUI difftool: \
`$ git pr show -t 18821`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18821.diff">https://git.openjdk.org/jdk/pull/18821.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18821#issuecomment-2067120255)

### CODE REVIEW

File: src/hotspot/os/bsd/os_bsd.cpp [hunk 1]
```diff
@@ -2133,23 +2133,32 @@ jint os::init_2(void) {
   FLAG_SET_ERGO(UseNUMAInterleaving, false);
 
   if (MaxFDLimit) {
-    // set the number of file descriptors to max. print out error
+    // Set the number of file descriptors to max. print out error
```
Comment (dcubed-ojdk) [inline src/hotspot/os/bsd/os_bsd.cpp:2136]: You dropped the other editorial fixed, but only kept part of this one. In the previous
version of patch (8300088), you also fixed: s/print/Print/

I would drop this editorial fix also.

**Review by dcubed-ojdk — APPROVED** [review]
I compared this patch to the previous patch (8300088) and it is the
same in the core part of the fix. All but one of the editorial changes
from 8300088 have been dropped which is good for a backport.

It would be good if you revived all of the editorial fixes from 8300088
and integrated them into the main line using a separate RFE.

Thanks for documenting your testing.

**Review by dholmes-ora — APPROVED** [review]
LGTM2

Thanks

File: src/hotspot/os/bsd/os_bsd.cpp (hunk 1 above)
Comment (gerard-ziemski) [reply src/hotspot/os/bsd/os_bsd.cpp]: Done.

**Comment by gerard-ziemski** [comment]
@dcubed-ojdk @dholmes-ora  re-requested reviews.

For such a trivial change (1 letter from capital to small in a comment) not sure we need it, but it's a protocol, so I'm going to wait for your re-reviews.

Thanks!

**Pull request closed without merging**